cli/cmd/context/ls.go

```diff
@@ -93,35 +95,41 @@ func runList(cmd *cobra.Command, opts lsOpts) error {
 	}
 
 	if opts.json {
-		j, err := formatter.ToStandardJSON(contexts)
-		if err != nil {
-			return err
-		}
-		fmt.Println(j)
-		return nil
+		opts.format = formatter.JSON
 	}
 
-	w := tabwriter.NewWriter(os.Stdout, 20, 1, 3, ' ', 0)
-	fmt.Fprintln(w, "NAME\tTYPE\tDESCRIPTION\tDOCKER ENDPOINT\tKUBERNETES ENDPOINT\tORCHESTRATOR")
-	format := "%s\t%s\t%s\t%s\t%s\t%s\n"
+	return printContextLsFormatted(opts.format, currentContext, os.Stdout, contexts)
+}
 
-	for _, c := range contexts {
-		contextName := c.Name
-		if c.Name == currentContext {
-			contextName += " *"
+func printContextLsFormatted(format string, currContext string, out io.Writer, contexts []*store.DockerContext) error {
+	var err error
+	switch strings.ToLower(format) {
```

Comment:
What about
```golang
err = formatter.Print(context, format, func(w io.Writer) {
	for _, c := range contexts {
		contextName := c.Name
		if c.Name == currContext {
			contextName += " *"
		}
		fmt.Fprintf(w, "%s\t%s\t%s\t%s\t%s\t%s\n",
			contextName,
			c.Type(),
			c.Metadata.Description,
			getEndpoint("docker", c.Endpoints),
			getEndpoint("kubernetes", c.Endpoints),
			c.Metadata.StackOrchestrator)
	}
} , "NAME", "TYPE", "DESCRIPTION", "DOCKER ENDPOINT", "KUBERNETES ENDPOINT", "ORCHESTRATOR")
```

And we let the formatter switch the format and call the `ToStandardJSON` when needed?
This way this can be reused everywhere.

Reply:
+1. This goes with my concerns above regarding exposing our go structs and not returning the same information depending on format json or pretty. Using the formatter to output both json and pretty enforces better choices there

Reply:
Done